### PR TITLE
Add support for dynamic internal repo path

### PR DESCRIPTION
**Pull request by irosenzw**
Adding support for creating an 'internal repo' folder not in the default location: "deployment-.../default/internal_repo/"

Signed-off-by: Ido Rosenzwig <irosenzw@redhat.com>

### CODE REVIEW

**Comment by eedri** [comment]
ci test please

File: lago/prefix.py [hunk 1]
```diff
@@ -104,6 +104,14 @@ def __init__(self, prefix):
         self.paths = paths.Paths(self._prefix)
         self._virt_env = None
         self._metadata = None
+        self._internal_repo = None
+
+    def _get_internal_repo (self):
+        return self._internal_repo
+
+    def _set_internal_repo (self, path):
+        self._internal_repo = path
```
Comment (david-caro) [inline lago/prefix.py]: this is not needed, there's no extra access logic or anything. This is not java ;)

File: lago/ssh.py [hunk 2]
```diff
@@ -354,7 +354,7 @@ def get_ssh_client(
                 )
 
             ssh_tries -= 1
-            time.sleep(1)
+            time.sleep(5)
```
Comment (david-caro) [inline lago/ssh.py]: seems unrelated?

File: ovirtlago/__init__.py [hunk 3]
```diff
@@ -210,15 +210,24 @@ def _create_rpm_repository(
         repo_names,
         custom_sources=None,
         projects_list=None,
+        dist_out=None,
```
Comment (david-caro) [inline ovirtlago/__init__.py]: this name is misleading, the param that's called 'dists' is for 'distributions', as in fc23/el7. While here dist_out dose seem to be an output path?

File: ovirtlago/__init__.py [hunk 4]
```diff
@@ -210,15 +210,24 @@ def _create_rpm_repository(
...
             projects_list = PROJECTS_LIST
 
+
         custom_sources = custom_sources or []
 
         rpm_dirs = []
+
+        dist_output = self.paths.internal_repo()
+        if (dist_out):
+            dist_output = dist_out
```
Comment (david-caro) [inline ovirtlago/__init__.py]: it's nicer to do both in the same line, probably without creating an extra var, a 'dummy' example:

`dist_out = dist_out or self.paths.internal_repo()`

The above is readable, but does match empty stings and the like (anything python considers false), if you want to allow them you should use:

`dist_out = dist_out if dist_out is not None else self.paths.internal_repo()`

Comment (david-caro) [inline ovirtlago/__init__.py]: Also, I'd put the assignment at the start of the function (or closer to where it's used)

File: ovirtlago/__init__.py [hunk 5]
```diff
@@ -249,8 +259,10 @@ def prepare_repo(
         rpm_repo=None,
         reposync_yum_config=None,
         skip_sync=False,
-        custom_sources=None
+        custom_sources=None,
+        dist_out=None,
     ):
+        _dist_out = dist_out
```
Comment (david-caro) [inline ovirtlago/__init__.py]: this extra assignment seems to do nothing?

File: ovirtlago/cmd.py [hunk 6]
```diff
@@ -114,6 +114,11 @@ def do_ovirt_runtest(prefix, test_file, **kwargs):
     type=os.path.abspath,
 )
 @cli_plugin_add_argument(
+    '--repo-path',
```
Comment (david-caro) [inline ovirtlago/cmd.py]: I'd be a little bit more explicit, something like '--internal-repo-path' or '--prefix-repo-path'. There's another option that has already a bad naming just before '--rpm-repo' that makes this one confusing :confused:

File: ovirtlago/cmd.py [hunk 7]
```diff
@@ -157,6 +162,7 @@ def do_ovirt_reposetup(
     if kwargs['ioprocess_dir']:
         warnings.warn('Deprecated option --ioprocess-dir ignored')
 
+    _repo_path = kwargs['repo_path']
```
Comment (david-caro) [inline ovirtlago/cmd.py]: as this is a non-deprecated option, is better if you declare it as a parameter of the function (just as prefix or rpm_repo), that way you don't need to access the kwargs yourself

**Comment by david-caro** [comment]
I'm curious :smile:, what's your use case?

**Comment by irosenzw** [comment]
Open new PR: https://github.com/lago-project/lago/pull/287
